### PR TITLE
V4 develop

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

File: src/constants/protocols.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export const ANALOG = 'analog';
 export const MTX = 'mtx';
```
Comment (DarkPark) [inline src/constants/protocols.js:2]:
```suggestion
export const MTX1 = 'mtx1';
```

should we switch to this?

File: src/utils/getBytesFromString.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import {getBytesFromHex, getBytesFromBase64} from '@jooby-dev/jooby-codec/utils/index.js';
+import {HEX} from '@jooby-dev/jooby-codec/constants/bytesConversionFormats.js';
+
+
+const getBytesFromString = ( bytes, bytesConversionFormats = HEX ) => (bytesConversionFormats === HEX
```
Comment (DarkPark) [inline src/utils/getBytesFromString.js]:
```suggestion
const getBytesFromString = ( bytes, bytesConversionFormats = HEX ) => (
    bytesConversionFormats === HEX
```
?

File: src/utils/getStringFromBytes.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import {getHexFromBytes, getBase64FromBytes} from '@jooby-dev/jooby-codec/utils/index.js';
+import {BASE64} from '@jooby-dev/jooby-codec/constants/bytesConversionFormats.js';
+
+
+const getStringFromBytes = ( bytes, {bytesConversionFormat} = {} ) => (
+    bytesConversionFormat === BASE64 ? getBase64FromBytes(bytes) : getHexFromBytes(bytes)
```
Comment (DarkPark) [inline src/utils/getStringFromBytes.js]:
```suggestion
    bytesConversionFormat === BASE64
        ? getBase64FromBytes(bytes) : getHexFromBytes(bytes)
```

?

File: readme.md [hunk 4]
```diff
@@ -78,12 +74,10 @@ Available environment variables:
 | `POST` | `/v1/decoder`              | General decoder route. Requires to specify [protocol](#protocols) in the requests body.  |
```
Comment (DeadByElpy) [inline readme.md]:
```suggestion
| `POST` | `/v2/decoder`              | General decoder route. Requires to specify [protocol](#protocols) in the requests body.  |
```

and others

**Review by DeadByElpy — CHANGES_REQUESTED** [review]

File: src/routes/v2/index.js [hunk 5]
```diff
@@ -14,10 +13,9 @@ export default async fastify => {
     });
 
     analog(fastify);
-    chirpstack(fastify);
+    //chirpstack(fastify);
```
Comment (DarkPark) [inline src/routes/v2/index.js]: why disable?

File: src/constants/protocols.js (hunk 1 above)
Comment (Pnom) [reply src/constants/protocols.js:2]: use names same as in jooby-codec

File: src/utils/getBytesFromString.js (hunk 2 above)
Comment (Pnom) [reply src/utils/getBytesFromString.js]: fixed

File: src/utils/getStringFromBytes.js (hunk 3 above)
Comment (Pnom) [reply src/utils/getStringFromBytes.js]: fixed

File: src/routes/v2/index.js (hunk 5 above)
Comment (Pnom) [reply src/routes/v2/index.js]: fixed

File: readme.md (hunk 4 above)
Comment (Pnom) [reply readme.md]: fixed

**Review by DeadByElpy — APPROVED** [review]

File: src/utils/getStringFromBytes.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import {getHexFromBytes, getBase64FromBytes} from '@jooby-dev/jooby-codec/utils/index.js';
+import {BASE64} from '@jooby-dev/jooby-codec/constants/bytesConversionFormats.js';
+
+
+const getStringFromBytes = ( bytes, {bytesConversionFormat} = {} ) => (
+    bytesConversionFormat === BASE64
+        ? getBase64FromBytes(bytes) : getHexFromBytes(bytes)
```
Comment (DarkPark) [inline src/utils/getStringFromBytes.js]:
```suggestion
        ? getBase64FromBytes(bytes)
        : getHexFromBytes(bytes)
```

File: src/utils/getBytesFromString.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import {getBytesFromHex, getBytesFromBase64} from '@jooby-dev/jooby-codec/utils/index.js';
+import {HEX} from '@jooby-dev/jooby-codec/constants/bytesConversionFormats.js';
+
+
+const getBytesFromString = ( bytes, bytesConversionFormats = HEX ) => (
+    bytesConversionFormats === HEX
+        ? getBytesFromHex(bytes) : getBytesFromBase64(bytes)
```
Comment (DarkPark) [inline src/utils/getBytesFromString.js]:
```suggestion
        ? getBytesFromHex(bytes)
        : getBytesFromBase64(bytes)
```

Comment (Pnom) [reply src/utils/getBytesFromString.js]: fixed

File: src/utils/getStringFromBytes.js (hunk 6 above)
Comment (Pnom) [reply src/utils/getStringFromBytes.js]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**